src/common/components/Input.common.tsx

```diff
@@ -0,0 +1,47 @@
+import { ChangeEvent } from 'react';
+import {
+  Path,
+  RegisterOptions,
+  FormState,
+  UseFormRegister,
+} from 'react-hook-form';
+
+import classes from '../../common/components/Input.module.scss';
+
+export const _randomBackground = (evt: ChangeEvent<HTMLInputElement>) => {
+  const classNames = Object.keys(classes);
+  const randomIndex = Math.floor(classNames.length * Math.random());
+  const randomClass = classes[classNames[randomIndex]];
+  const currentClasses = evt.currentTarget.classList;
+  currentClasses.forEach((cls) => currentClasses.remove(cls));
+  currentClasses.add(classes.input);
+  currentClasses.add(randomClass);
+};
+
+export const inputRegisterFactory = <T,>({
```

Comment:
I'm still not sure how I feel about this, it seems... hacky

Reply:
Looking at https://react-hook-form.com/api/useform, validation occurs by default on onChange events, which means it will only validate the contents of a form if they are changed. That seems like preferred behavior over this perhaps? I'm not clear what benefit we get from this

Reply:
The `validated` prop is meant to indicate whether or not the input has been validated and the errors are the errors for all the fields. The errors are only passed to a component if there is an error for that named field. The alternative to this is to do this logic when props are passed to the <Input /> component, but you want to do the same thing each time, so I put it in the factory to keep things DRY.